src/modules/sni/item.cpp

```diff
@@ -318,19 +314,42 @@ void Item::updateImage() {
 }
 
 Glib::RefPtr<Gdk::Pixbuf> Item::getIconPixbuf() {
-  try {
-    if (!icon_name.empty()) {
+
+  if (!icon_name.empty()) {
+
+    try {
       std::ifstream temp(icon_name);
       if (temp.is_open()) {
         return Gdk::Pixbuf::create_from_file(icon_name);
       }
+    } catch (Glib::Error& e) {
+      // Ignore because we want to also try different methods of getting an icon.
+      //
+      // But a warning is logged, as the file apparently exists, but there was
+      // a failure in creating a pixbuf out of it.
+
+      spdlog::warn("Item '{}': {}", id, static_cast<std::string>(e.what()));
+    }
+
+    try {
+      // Will throw if it can not find an icon.
       return getIconByName(icon_name, getScaledIconSize());
-    } else if (icon_pixmap) {
-      return icon_pixmap;
+    } catch (Glib::Error& e) {
+      spdlog::info("Item '{}': {}", id, static_cast<std::string>(e.what()));
```

Comment:
Not sure if `info` is appropriate. This is most likely just noise; `trace` might(?) be more appropriate.

I also could not find a documentation listing all the possible exceptions, which then could be used to differentiate between "errors" and "icon not found".

Reply:
Replace with trace and i'll merge the PR :)